components/script/dom/serviceworkerglobalscope.rs

```diff
@@ -164,7 +164,7 @@ impl ServiceWorkerGlobalScope {
                 destination: Destination::ServiceWorker,
                 credentials_mode: CredentialsMode::Include,
                 use_url_credentials: true,
-                origin: script_url,
+                origin: GlobalScope::current().origin().immutable().clone(),
```

Comment:
Same problem as the other worker code - we need to get the origin before spawning the new thread.